pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,243 @@
+    author = "Xanadu Inc."
+
+    supported_devices = {
+        "default.qubit",
+        "default.qubit.tf",
+        "default.qubit.jax",
+        "default.qubit.torch",
+        "lightning.qubit",
+    }
+
+    def __init__(
+        self,
+        wires,
+        *,
+        shots=None,
+        analytic=None,
+        backend="default.qubit",
+        gen_report: Union[bool, str] = False,
+        future=False,
+    ):
+        super().__init__(0, shots, cache=False)
+
+        self._backend = backend
+        self._backend_cls = qml.plugin_devices[backend].load()
+        self._backend_dev = self._backend_cls(wires=0)
+        self._wires = wires
+        self._gen_report = gen_report
+        self.num_wires = wires if isinstance(wires, int) else len(wires)
+        self._future = future
+        self.expand_fn = self.default_expand_fn
+
+        if backend not in TaskQubit.supported_devices:
+            raise Exception("Unsupported device backend.")
+
+    def __get__(self, obj, objtype=None):
+        if obj not in self.__dict__:
+            return self._backend_cls.obj
+        return self.__dict__[obj]
+
+    def batch_execute(self, circuits: List[qml.tape.QuantumTape]):
+        "This overloads the backt_execute functionality of QubitDevice to offload computations to a user-chosen backend device. This allows scaling of the available workers to instantiate a given number of backends for concurrent circuit evaluations."
+        if self._gen_report:
+            filename = self._gen_report if isinstance(self._gen_report, str) else "dask-report.html"
+            cm = performance_report(filename=filename)
+        else:
+            cm = nullcontext()
+
+        with cm:
+            results = []
+            if isinstance(circuits, dask.distributed.client.Future):
+                with worker_client() as client:
+                    results = client.submit(
+                        lambda backend, wires, tapes: [
+                            TaskQubit._execute_wrapper(backend, wires, i) for i in tapes
+                        ],
+                        self._backend,
+                        self._wires,
+                        circuits,
+                    )
+            else:
+                with worker_client() as client:
+                    for circuit in circuits:
+                        results.append(
+                            client.submit(
+                                TaskQubit._execute_wrapper,
+                                self._backend,
+                                self._wires,
+                                circuit,
+                            )
+                        )
+
+            if self._future:
+                return results
+            with worker_client() as client:
+                res = client.gather(results)
+            return res
+
+    def apply(self, operations, **kwargs):
+        "The apply method of task.qubit should not be explicitly used."
+        pass
```

Comment:
is this because we currently force it to be abstract?

Reply:
Yep, that's right.